Moving Data Analysis GUI interface files to inelastic folder.

adriazalvarez opened the pull request:
### Description of work
A new qt library / test library for the inelastic interface category has been created.
 A new inelastic subdirectory has been created,  and the data analysis interface files and test files have been moved 
to the new category.
#### Summary of work
<!-- Please provide a short, high level description of the work that was done.
-->

<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
#### Purpose of work
This can be removed if a github issue is referenced below
-->

Fixes #35665 
Fixes #36323

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has been undertaken.
-->

### To test:

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

adriazalvarez committed:
moved relevant files to inelastic/data analysis folder, added interface dll

adriazalvarez committed:
inelastic cmake

adriazalvarez committed:
changes to eliminate build errors

adriazalvarez committed:
moving test files/adding new test interface

adriazalvarez committed:
Fix typo in cmake file

adriazalvarez committed:
verbatim last commit

robertapplin reviewed (changes requested):
Code changes look good and the hard part (creating a qt library) is done!

I've just realised how many Data Analysis files there are. We also need to moves these and their related test files:

- [ ] `IFQFitObserver`
- [ ] `IndirectDockWidgetArea`
- [ ] `IndirectFitPlotModel`
- [ ] `IndirectFitPlotView`
- [ ] `IndirectFitPropertyBrowser`
- [ ] `ConvFitModel`
- [ ] `FQFitConstants`
- [ ] `FitStatusWidget`
- [ ] `FunctionTemplateBrowser`
- [ ] `IDAFunctionParameterEstimation`
- [ ] `IIndirectFittingModel`
- [ ] `IIndirectFitDataView`
- [ ] `IIndirectFitOutput`
- [ ] `IIndirectFitOutputOptionsModel`
- [ ] `IIndirectFitOutputOptionsView`
- [ ] `IIndirectFitPlotView`
- [ ] `IIndirectFitPlotModel`
- [ ] `IIndirectFitPlotPresenter`
- [ ] `IIndirectFittingModel`
- [ ] `IndirectEditResultsDialog`
- [ ] `IndirectFitData`
- [ ] `IndirectFitDataModel`
- [ ] `IndirectFitDataPresenter`
- [ ] `IndirectFitDataView`
- [ ] `IndirectFitOutput`
- [ ] `IndirectFitOutputOptionsPresenter`
- [ ] `IndirectFitTab`
- [ ] `IndirectSpectrumSelector`
- [ ] `ParameterEstimation`

adriazalvarez committed:
Remaining files of data analysis + data manipulation files (+corresponding tests)

adriazalvarez commented:
I have added the remaining files in the last commit. 
Sorry for the huge file dump. It was very entangled with data analysis and data manipulation, 
so I have decided to merge this issue with #36323 . This last commit makes part of that. 
(at the end there were data manipulation classes calling data analysis classes). Additionally, I had to move the indirect function browser folder to inelastic folder to resolve dependencies in data analysis.
Finally I also moved the test.

robertapplin commented on qt/scientific_interfaces/Indirect/CMakeLists.txt:
```diff
@@ -226,38 +103,24 @@ set(UI_FILES
-    IndirectEditResultsDialog.ui
-    IndirectFitDataView.ui
-    IndirectFitOutputOptions.ui
     IndirectFitPreviewPlot.ui
```
I think we want to move this to the Inelastic/Analysis folder

robertapplin commented on qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h:
```diff
@@ -24,7 +24,7 @@ namespace CustomInterfaces {
 
 // Used for painting an Icon onto the handle of the splitter on workbench
-class SplitterHandle : public QSplitterHandle {
+class MANTIDQT_INELASTIC_DLL SplitterHandle : public QSplitterHandle {
```
I think we probably don't need the export macro on this class

robertapplin commented on qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h:
```diff
@@ -41,7 +41,7 @@ class SplitterHandle : public QSplitterHandle {
 };
 
-class Splitter : public QSplitter {
+class MANTIDQT_INELASTIC_DLL Splitter : public QSplitter {
```
And same with this class?

robertapplin commented on qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPropertyBrowser.h:
```diff
@@ -38,10 +38,10 @@ namespace IDA {
 
-class FunctionTemplateBrowser;
-class FitStatusWidget;
+class MANTIDQT_INELASTIC_DLL FunctionTemplateBrowser;
+class MANTIDQT_INELASTIC_DLL FitStatusWidget;
```
These are just forward declarations and so will not need the export macro
https://stackoverflow.com/questions/4757565/what-are-forward-declarations-in-c

adriazalvarez committed:
removing unnecesary dllexport macros

robertapplin reviewed (changes requested):
This new file organisation is much better, thanks! Can you rename the `IndirectFunctionBrowser` folder to `FunctionBrowser`, and move it into the Analysis folder?

robertapplin commented on qt/scientific_interfaces/Indirect/IndirectFitDataModel.h:
```diff
@@ -23,7 +23,7 @@ using namespace MantidWidgets;
    IndirectFitDataModel - Specifies an interface for updating, querying and
    accessing the raw data in IndirectDataAnalysisTabs
 */
```
I think you might have missed removing this file from the Indirect folder? It is already in the Inelastic/Analysis folder

adriazalvarez committed:
New FunctionBrowser folder, remove unnecesary dll exports

adriazalvarez committed:
Resolve some dependencies for inelastic test

adriazalvarez committed:
Adding cpp suppression shadow function:FqFitAddworkspaceDialog

adriazalvarez committed:
Fixing tests

adriazalvarez committed:
fixing ui dependencies in new interface

robertapplin reviewed (approved):
Thanks for all your hard work on this. This is a really good first step for organising the files in Indirect/Inelastic. It lowers the entry threshold for new starters, and also should make it easier for us to spot commonly used functionality that can be shared across interfaces in the future
